server/src/main/scala/drt/server/feeds/lgw/LGWForecastFeed.scala

```diff
@@ -0,0 +1,219 @@
+    val rows = theData.split("\n")
+    if (rows.length < 1) throw new Exception(s"The latest forecast file '$fileName' has no data.")
+    val header = rows.head
+    log.debug(s"The header of the CSV file $fileName is: '$header'.")
+    if (header.split(",").size != TOTAL_COLUMNS) {
+      log.warn(s"The CSV file header has does not have ${TOTAL_COLUMNS}, This is the header [$header].")
+    }
+    val body = rows.tail.filterNot(row => StringUtils.isBlank(row.replaceAll(",", "")))
+    log.debug(s"The latest forecast file has ${body.length} rows.")
+    body.flatMap(toArrival).toList
+  }
+
+  private def toArrival(row: String): Option[Arrival] = Try {
+
+    val fields = regex.findAllIn(row).map(field => StringUtils.removeEnd(StringUtils.removeStart(StringUtils.removeEnd(field, ","), "\"" ), "\"")).toList
+
+    def scheduledDateAsIsoString = Try {
+      val dateTimeField = StringUtils.trimToEmpty(fields(DATE_TIME))
+      `dd/mm/yyyy HH:mm`.parseDateTime(dateTimeField).toString(ISODateTimeFormat.dateTime)
+    } match {
+      case Success(value) => value
+      case Failure(exception) => throw new Exception(s"""Cannot get the scheduled date from "$row".""", exception)
+    }
+
+    val scheduledDate = scheduledDateAsIsoString
+
+    new Arrival(Operator = "",
+      Status = PORT_FORECAST,
+      EstDT = "",
+      ActDT = "",
+      EstChoxDT = "",
+      ActChoxDT = "",
+      Gate = "",
+      Stand = "",
+      MaxPax = fields(MAX_PAX).toInt,
+      ActPax = fields(ACTUAL_PAX).toInt,
+      TranPax = fields(TRANSFER_PAX).toInt,
+      RunwayID = "",
+      BaggageReclaimId = "",
+      FlightID = 0,
+      AirportID = LGW,
+      Terminal = fields(5) match {
+        case "South" => "S"
+        case "North" => "N"
+        case unknown => throw new Exception(s"Unknown Terminal in Gatwick Forecast feed $unknown.")
+      },
+      rawICAO = fields(FLIGHT_NUMBER),
+      rawIATA = fields(FLIGHT_NUMBER),
+      Origin = fields(AIRPORT_CODE),
+      SchDT = scheduledDate,
+      Scheduled = SDate.parseString(scheduledDate).millisSinceEpoch,
+      PcpTime = 0,
+      None)
+  } match {
+    case Success(arrival) => Some(arrival)
+    case Failure(error) =>
+      log.error(s"""Cannot parse arrival from "$row".""", error)
+      None
+
+  }
+
+  private def downloadTheData(boxAPIConnection: BoxAPIConnection, latestFile: BoxFile#Info): String =
+    Try {
+      val file = new BoxFile(boxAPIConnection, latestFile.getID)
+      val stream = new ByteArrayOutputStream()
+      file.download(stream)
+      stream.flush()
+      stream.close()
+      new String(stream.toByteArray, "UTF-8")
+    } match {
+      case Success(string) => string
+      case Failure(e: BoxAPIResponseException) => log.error(e.getResponse, e)
+        throw e
+      case Failure(exception) => throw exception
+    }
+
+  private def getTheLatestFileInfo(boxAPIConnection: BoxAPIConnection): BoxFile#Info = {
```

Comment:
Same for this, return a Try here, otherwise these Exceptions aren't being handled that I can see.